File: app/src/main/java/io/duckduckgosearch/app/DuckAutoCompleteTextView.java
package io.duckduckgosearch.app;

import android.annotation.SuppressLint;
import android.content.Context;
import android.graphics.Rect;
import android.util.AttributeSet;
import android.widget.AutoCompleteTextView;

@SuppressLint("AppCompatCustomView")
public class DuckAutoCompleteTextView extends AutoCompleteTextView {

    public DuckAutoCompleteTextView(Context context) {
        super(context);
    }

    public DuckAutoCompleteTextView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public DuckAutoCompleteTextView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public int getThreshold() {
        return 0;
    }

    @Override
    public boolean enoughToFilter() {
        return true;
    }

    @Override
    protected void onFocusChanged(boolean focused, int direction, Rect previouslyFocusedRect) {
        super.onFocusChanged(focused, direction, previouslyFocusedRect);
        if (focused && getAdapter() != null) {
            ((AutoCompleteAdapter)getAdapter()).getFilter().filter(getText());
        }
    }
}

File: app/src/main/java/io/duckduckgosearch/app/DuckAutoCompleteTextView.kt
package io.duckduckgosearch.app

import android.annotation.SuppressLint
import android.content.Context
import android.graphics.Rect
import android.util.AttributeSet
import android.widget.AutoCompleteTextView

@SuppressLint("AppCompatCustomView")
class DuckAutoCompleteTextView : AutoCompleteTextView {
    constructor(context: Context?) : super(context)
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs)
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    override fun getThreshold(): Int {
        return 0
    }

    override fun enoughToFilter(): Boolean {
        return true
    }

    override fun onFocusChanged(focused: Boolean, direction: Int, previouslyFocusedRect: Rect?) {
        super.onFocusChanged(focused, direction, previouslyFocusedRect)
        if (focused && adapter != null) {
            (adapter as AutoCompleteAdapter).filter.filter(text)
        }
    }
}